app/src/main/java/com/criteo/testapp/InHouseActivity.java

```diff
@@ -62,18 +61,25 @@ protected void onCreate(Bundle savedInstanceState) {
     setContentView(R.layout.activity_in_house);
     MockedIntegrationRegistry.force(Integration.IN_HOUSE);
 
-    adLayout = findViewById(R.id.AdLayout);
+    criteoBannerView = findViewById(R.id.criteoBannerView);
+    criteoBannerView.setCriteoBannerAdListener(new TestAppBannerAdListener(
+        TAG, "In-House"));
+
     btnShowInterstitial = findViewById(R.id.buttonInhouseInterstitial);
     btnShowInterstitialIbv = findViewById(R.id.buttonInhouseInterstitialIbv);
     context = getApplicationContext();
 
     nativeLoader = new CriteoNativeLoader(
         NATIVE,
-        new TestAppNativeAdListener(TAG, NATIVE.getAdUnitId(), adLayout),
+        new TestAppNativeAdListener(TAG, NATIVE.getAdUnitId()),
         new TestAppNativeRenderer()
     );
 
-    findViewById(R.id.buttonInhouseBanner).setOnClickListener(v -> loadBannerAd());
+    findViewById(R.id.buttonInhouseBanner).setOnClickListener(v -> {
+      Log.d(TAG, "Banner Requested");
+      Criteo.getInstance().loadBid(BANNER, criteoBannerView::loadAd);
```

Comment:
You have inlined the `loadBannerAd` but the method is still here. Personally I perfer to have really short lambda to have a more readable code at call site. So I would suggest to revert this line.